qiskit_nature/problems/sampling/protein_folding/peptide/beads/base_bead.py

```diff
@@ -0,0 +1,76 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""An abstract class defining a bead of a peptide."""
+from abc import ABC
+from typing import List, Tuple, Union
+
+from qiskit.opflow import PauliOp, OperatorBase
+
+from ..pauli_ops_builder import _build_full_identity
+from ...residue_validator import _validate_residue_symbol
+
+
+class BaseBead(ABC):
+    """An abstract class defining a bead of a peptide."""
+
+    def __init__(
+        self,
+        chain_type: str,
+        main_index: int,
+        residue_type: str,
+        turn_qubits: List[PauliOp],
+        _build_turn_indicator_fun_0,
+        _build_turn_indicator_fun_1,
+        _build_turn_indicator_fun_2,
+        _build_turn_indicator_fun_3,
+    ):
```

Comment:
Missing docstring and the latter params - why start with _ and missing typehints